components/global.test.js

```diff
@@ -0,0 +1,97 @@
+import { shallow } from "enzyme";
+import   {MyApp } from "../pages/_app"
+import AppHead from "../Utils/layouts/head";
+import { PageFooter} from "../components/footer"
+import BreadCrumb, {BreadAnchor, BreadLink} from "../Utils/layouts"
+import * as nextRouter from 'next/router';
+
+
+describe("AppJs", () => {
```

Comment:
There are four components that make up `AppJs`: `Head`, `Header`, `Container` and `Footer`.
- Would it make sense to test each of these in this section? Currently there are only tests for 2 of these here.
- The header component is tested within the header directory, which is inconsistent with how the footer component is dealt with. Could we think about making this structure more consistent?

Reply:
That was the idea. The global test is specifically target at AppJs and other components that are not directly related to any of the 2 pages. Then each of header and footer should have their own tests. What was tested in AppJs was  to check if it renders the header and the footer, which does not mean testing the header and the footers themselves because it is expected that those would have been tested in the library. The only reason we would want to test any of those again would be if there are additional implementation in the app as we have it in header, where we added the breadcrumbs (which was not originally in the library). As it is, the footer was used as delivered by the library, unless I want to check if it has some attributes. I hope I am able to answer your question. I would welcome any idea that will improve the consistency of the project structure.

Reply:
OK, sounds good! 👍